From e389e61d2be1ca0bb1682573e7292865cee2feaa Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 18 Sep 2022 12:24:05 +0300
Subject: [PATCH] [SW-104157] gaudi2: pll: init plls for fast sram access

SRAM self repair and scrubbing are performed in ppboot. In order
to speed up the SRAM access time, the relevant PLLs are initialized
to nominal frequency.

Change-Id: I76b345502ee137509e16b3a2a4ca3daf8f3b33f9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi2/common/pll.c b/arch/gaudi2/common/pll.c
index 7f691ee..0c94277 100644
--- a/arch/gaudi2/common/pll.c
+++ b/arch/gaudi2/common/pll.c
@@ -36,6 +36,81 @@
 
 static struct hl_pll_cfg gaudi2_sram_pll_cfg[] = {
 	{
+		.name = "PCI_PLL",
+		.base = mmDCORE0_PCI_PLL_CTRL_BASE,
+		.nr = 1, .nf = 60, .od = 3, .nb = 2,	/* 500 */
+		/*
+		 * div 0 div_factor differ from spec due to CDC workaround required for
+		 * pcie link init. it is set to nominal clk by preboot after the pcie link
+		 * is up.
+		 */
+		.div_factor = { 4, -1, 4, 4 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE0_IF",
+		.base = mmDCORE0_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE1_IF",
+		.base = mmDCORE1_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE2_IF",
+		.base = mmDCORE2_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE3_IF",
+		.base = mmDCORE3_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE0_MESH",
+		.base = mmDCORE0_XBAR_MESH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE1_MESH",
+		.base = mmDCORE1_XBAR_MESH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
 		.name = "DCORE2_SRAM",
 		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
 		.nr = 1, .nf = 63, .od = 3, .nb = 3,	/* 350 */
